code/game/objects/structures/window.dm

```diff
@@ -363,11 +363,11 @@
 	move_update_air(T)
 
 /obj/structure/window/CanAtmosPass(turf/T)
-	if(get_dir(loc, T) == dir)
-		return !density
-	if(dir == FULLTILE_WINDOW_DIR)
-		return !density
-	return 1
+	if(!anchored || !density)
+		return TRUE
+	if(get_dir(loc, T) == dir || dir == FULLTILE_WINDOW_DIR)
+		return FALSE
+	return TRUE
```

Comment:
return !(get_dir(loc, T) == dir || dir == FULLTILE_WINDOW_DIR)

Reply:
In any other place I'd say that isn't worth the readability cost but yeah, this is a hot proc.